services/web/server/src/simcore_service_webserver/publication_handlers.py

```diff
@@ -0,0 +1,56 @@
+import json
+import os
+
+from aiohttp import MultipartReader, hdrs, web
+
+from .login.decorators import RQT_USEREMAIL_KEY, login_required
+
+support_email_address = 'support@osparc.io'
+email_template_name = 'service_submission.html'
+
+@login_required
+async def service_submission(request: web.Request):
+    reader = MultipartReader.from_response(request)
+    data = None
+    filedata = None
+    # Read multipart email
+    while True:
+        part = await reader.next()
+        if part is None:
+            break
+        if part.headers[hdrs.CONTENT_TYPE] == 'application/json':
+            data = await part.json()
+            continue
+        if part.headers[hdrs.CONTENT_TYPE] == 'application/zip':
+            filedata = await part.read(decode=True)
+            # Validate max file size
+            maxsize = 10 * 1024 * 1024 # 10MB
+            actualsize = len(filedata)
+            if actualsize > maxsize:
+                raise web.HTTPRequestEntityTooLarge(maxsize, actualsize)
+            filename = part.filename
+            continue
+        raise web.HTTPUnsupportedMediaType(reason=f'One part had an unexpected type: {part.headers[hdrs.CONTENT_TYPE]}')
+    # data (dict) and file (bytearray) have the necessary information to compose the email
+    user_email = request.get(RQT_USEREMAIL_KEY, -1)
+    try:
+        # send email
+        from .login.utils import (render_and_send_mail, common_themed)
+        from json2html import json2html
```

Comment:
does this need to be imported here?

Reply:
hmm I guess not... it is only used there though... I will move it to the top. my Python sucks

Reply:
oh yes. I put it there because otherwise I get ``RuntimeError: 1 handlers were not mapped to routes: dict_keys(['render_and_send_mail'])``. it tries to match it as if it was a route handler... :S

Reply:
discussed